.github/workflows/release-pypi.yml

```diff
@@ -0,0 +1,79 @@
+---
+name: Release to PyPI
+
+on:
+  release:
+    types: [created]
+
+jobs:
+  build_sdist:
+    name: Build source package
+    runs-on: 'ubuntu-latest'
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Install/Upgrade Python dependencies
+      run: python -m pip install --upgrade pip wheel
+
+    - name: Build and install Python source package
+      run: |
+        python setup.py sdist
+        python -m pip install dist/*.tar.gz
+
+    - name: Run tests
+      run: |
+        pip install pytest pytest-xdist
+        pytest -n auto test/
+
+    - uses: actions/upload-artifact@v2
+      with:
+        path: dist/*.tar.gz
+
+
+  build_bdist:
+    name: Build binary wheels
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        # Windows build fails
+        os: [ubuntu-latest, macos-latest]  #  windows-latest
+
+    steps:
+    - uses: actions/checkout@v2
+
+    # setup Python for cibuildwheel
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.x'
+
+    # for other architectures, see: https://cibuildwheel.readthedocs.io/en/stable/faq/#emulation
+    - name: Build wheels for CPython
+      uses: pypa/cibuildwheel@v2.1.1
+      env:
+        CIBW_SKIP: 'cp310-* pp*'  # 3.10 and pypy builds are broken
+        CIBW_ARCHS: 'auto64'  # only 64-bit
+        CIBW_TEST_REQUIRES: pytest pytest-xdist
+        CIBW_TEST_COMMAND: pytest -n auto {project}/test/
+
+    - uses: actions/upload-artifact@v2
+      with:
+        path: ./wheelhouse/*.whl
+
```

Comment:
If I understand correctly this makes the wheel available for download directly from github actions?
Nice! :)

Reply:
I think it does make the artifacts downloadable directly from github (I haven't tried it but that's my impression) but that's not the primary reason.

The main benefit is that you can put the sdist package build + test and the bdist package(s) build + test in separate parallel jobs, and only upload the artifacts (from a third job) once all of the jobs have finished.  But each job obviously has a fresh CI image so the artifacts need to be shared somehow. 

If you just uploaded the artifacts from each of the two jobs then you could create a situation where one job passes and starts uploading packages to PyPI before the other has completed all of its work successfully, and that wouldn't be good.


>This uploads artifacts from your workflow allowing you to share data between jobs and store data once a workflow is complete.

https://github.com/actions/upload-artifact